### PR TITLE
Invite new user to PokeBattle

### PR DESCRIPTION
**Pull request by gabrielaleal**
## Description

- Added an "Invite User" button to the home page;
- Created a page to Invite Users;
- Wrote validation in case the email entered is from an already existing user;
- Send email to invited user.
## Related card

[As a Trainer I can invite an opponent by email if they are not yet in the system](https://trello.com/c/N2wYMtnQ/12-as-a-trainer-i-can-invite-an-opponent-by-email-if-they-are-not-yet-in-the-system)
## Steps to reproduce
You must be logged in.
1. Click on the "Invite User" button at the home page;
2. Enter an email.
### Overview
- [X] I assure that I've tested the functionality as a real user
- [X] I assure that I've tested success and error cases
- [X] I assure that I've tested other parts of the system that may be impacted by my changes
- [x] I've written unit tests for this code
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [X] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Another type of change, e.g. documentation
## Checklist

- [ ] My change requires a change to the documentation and I've updated the documentation accordingly
- [ ] My change requires a change to the dependencies and I've updated the dependencies accordingly

### CODE REVIEW

File: backend/battles/utils/email.py [hunk 1]
```diff
@@ -17,3 +18,18 @@ def send_battle_result(battle):
             "opponent_team": battle.opponent.teams.filter(battle=battle.id).first(),
         },
     )
+
+
+def send_user_invite_to_pokebattle(user_invited_email, user_who_invited_email):
+    signup_path = reverse_lazy("signup")
+    signup_url = f"{settings.HOST}{signup_path}"
```
Comment (amandasavluchinske) [inline backend/battles/utils/email.py:25]: This was a good workaround but it's pretty error prone. Imagine the domain changes and you forget to update it in settings. You can get this absolute url by doing this: https://stackoverflow.com/questions/2345708/how-can-i-get-the-full-absolute-url-with-domain-in-django

File: backend/pokebattle/settings/base.py [hunk 2]
```diff
@@ -34,6 +34,8 @@ def base_dir_join(*args):
 
 ALLOWED_HOSTS = []
 
+HOST = config('HOST', default='http://localhost:8000')
+
```
Comment (amandasavluchinske) [inline backend/pokebattle/settings/base.py:38]: Remove this when you update the absolute URL pls

File: backend/users/managers.py [hunk 3]
```diff
@@ -15,3 +15,6 @@ def create_superuser(self, **kwargs):
         user.is_staff = True
         user.save(using=self._db)
         return user
+
+    def check_if_email_already_exists(self, email):
+        return super().get_queryset().filter(email=email).exists()
```
Comment (amandasavluchinske) [inline backend/users/managers.py:20]: Is this something you're going to be reutilizing within the context of user querysets or is it going to be useful just this once?

File: backend/users/views.py [hunk 4]
```diff
@@ -36,3 +40,20 @@ class UserLoginView(LoginView):
...
+
+
+class InviteUserView(LoginRequiredMixin, generic.FormView):
+    template_name = "invite_user.html"
+    form_class = InviteUserForm
+    success_url = reverse_lazy("invite-user")
+
+    def form_valid(self, form):
+        user_invited_email = form.cleaned_data["email"]
+        send_user_invite_to_pokebattle(user_invited_email, self.request.user.email)
```
Comment (amandasavluchinske) [inline backend/users/views.py:52]: When you update this function to the new URL getting manner, you could pass `self.request` here so you can maintain a lower number of function arguments and still have access to `self.request.user.email` and the build_uri method inside.

File: backend/users/managers.py (hunk 3 above)
Comment (gabrielaleal) [reply backend/users/managers.py:20]: At least for now I'm just using this once.

File: backend/users/views.py (hunk 4 above)
Comment (gabrielaleal) [reply backend/users/views.py]: I didn't pass `self.request` because I thought it wouldn't be as readable

**Comment by amandasavluchinske** [comment]
@gabrielaleal I happened to notice this PR is on my court. Please remember to assign it back to the reviewer when the PR can be reviewed again. :)

File: backend/users/managers.py (hunk 3 above)
Comment (amandasavluchinske) [reply backend/users/managers.py:20]: ok

File: backend/users/views.py (hunk 4 above)
Comment (amandasavluchinske) [reply backend/users/views.py]: You're right, but I meant that if you were changing the URL fetching inside to build_uri.

**Pull request merged**